### ISSUE TITLE
Entry point arguments limited to `--version` or none

### ISSUE THREAD

**Issue by Yewzir** [RockstarLang/rockstar2#1]
Currently, the application's entry point is only using the first command-line argument, if provided and equals `--version`.

```cs
public static void Main(string[] args) {
    switch (args.Length) {
        case > 1:
            Console.WriteLine("Usage: rockstar <program.rock>");
            Environment.Exit(64);
            break;
        case 1:
	    if (args[0] == "--version") {
	        Console.WriteLine(RockstarEnvironment.VERSION);
		Environment.Exit(0);
	    }
	    RunFile(args[0]);
	    break;
        default:
            Console.WriteLine($"Rockstar {RockstarEnvironment.VERSION} on {System.Runtime.InteropServices.RuntimeInformation.RuntimeIdentifier}.");
	    Console.WriteLine("Type 'exit' to exit.");
	    RunPrompt();
	    break;
    }
}
```

Only the first array index is ever accessed, and the way I see it, the first case should be parsing additional command-line arguments that were supplied by the user, but a reference to the array is nowhere to be found.

One's usage expectation would generally resemble `rockstar <program.rock> <args>`, wouldn't it?

Regards
@Yewzir

**Comment by dylanbeattie** [comment]
That's a fair comment, but there's actually no support in the Rockstar language for command-line arguments at the moment - mainly 'cos this is the first time a native CLI version has really been a viable option. 

It shouldn't be that hard to plug in, but it would mean adding new syntax to the language. Out of curiosity, what would one's usage expectation be with respect to accessing those command line arguments from within the program?

**Comment by Yewzir** [comment]
Let's just consider Rockstar's entry point.

```cs
public static void Main(string[] args)
```
`Main` takes an _array_ as parameter, an _array_ of strings. Looking at your `switch` statement, there are only **two** references to that particular array. Both times, you index into the **first** element. And what about the rest, if any? I mean, it's an _array_... but when we fill it up with more data, we're using the application "incorrectly". But we aren't? It's an _array_, a _col-lec-tion_ of data.
```cs
case 1:
if (args[0] == "--version") {
	Console.WriteLine(VERSION);
	Environment.Exit(0);
}
RunFile(args[0]);
break;
```
May I kindly suggest you take a look at the [C#](https://rosettacode.org/wiki/Command-line_arguments#C#) snippets, and maybe some additional examples such as [Java](https://rosettacode.org/wiki/Command-line_arguments#Java), please? There're many examples of their uses in command-line interfaces.

Regards
@Yewzir

**Comment by duckyluuk** [comment]
I think what @Yewzir is trying to achieve is to have the command-line arguments usable as an input to the program. A possible implementation would be to simply write the remaining arguments into an array as a constant. Perhaps there could be a more rock-like name for this than just `args` / `arguments`. Maybe something like `playlist`, or something more arbitrary like `the words` could be a good fit.. Just spitting out ideas though.

**Comment by dylanbeattie** [comment]
Yeah, that makes sense. I'll come up with a suitably rock'n'roll syntax for it and get this added before 2.0 ships. Thanks :)

**Comment by Yewzir** [comment]
@dylanbeattie
Thank you very much for resolving our issue. The Rockstar project skyrockets with it.
